utils/client.ts

```diff
@@ -8,39 +8,43 @@ import { publicProvider } from 'wagmi/providers/public';
 import { SafeConnector } from 'wagmi/connectors/safe';
 import { optimism } from 'wagmi/chains';
 
-declare global {
-  interface Window {
-    rpcURL?: string;
-  }
-}
-const rpcURL = typeof window !== 'undefined' ? window?.rpcURL : undefined;
-
 export const walletConnectId = '11ddaa8aaede72cb5d6b0dae2fed7baa';
 
 const networkId = Number(process.env.NEXT_PUBLIC_NETWORK ?? optimism.id);
 export const defaultChain = Object.values(wagmiChains).find((c) => c.id === networkId) ?? optimism;
 
-const sortedChains = [defaultChain, ...Object.values(wagmiChains).filter((c) => c.id !== defaultChain.id)];
+export const isE2E: boolean = JSON.parse(process.env.NEXT_PUBLIC_IS_E2E ?? 'false');
 
-const alchemyKey = process.env.NEXT_PUBLIC_ALCHEMY_API_KEY;
+const sortedChains = isE2E
+  ? [defaultChain]
+  : [defaultChain, ...Object.values(wagmiChains).filter((c) => c.id !== defaultChain.id)];
 
-export const isE2E: boolean = JSON.parse(process.env.NEXT_PUBLIC_IS_E2E ?? 'false');
+const alchemyKey = process.env.NEXT_PUBLIC_ALCHEMY_API_KEY;
 
-const providers: ChainProviderFn<Chain>[] =
-  isE2E && rpcURL
-    ? [jsonRpcProvider({ rpc: () => ({ http: rpcURL }) })]
-    : [
-        ...(alchemyKey ? [alchemyProvider({ apiKey: alchemyKey })] : []),
-        publicProvider(),
-        w3mProvider({ projectId: walletConnectId }),
-      ];
+const providers: ChainProviderFn<Chain>[] = isE2E
+  ? [
+      jsonRpcProvider({
+        rpc: () => ({
+          http:
+            typeof window !== 'undefined'
+              ? // eslint-disable-next-line @typescript-eslint/no-explicit-any
+                (window as any).ethereum.provider.connection.url
```

Comment:
if you want to avoid casting window as any you can
```
declare global {
    interface Window { ethereum: {provider: connection: {url: string}}; }
}
```